ls1intum/Artemis
# `Development`: Improve background image input validation

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
Several endpoints in the classes ExamResource, ExerciseGroupResource and QuizExerciseResource do not sanitise the background file path of imported quiz exercises, which can lead to path traversal when being used without sanatisation.

### Description
We now:
1. Sanatize every background file path
2. Check if every background file path starts with `/api/files/drag-and-drop/backgrounds`

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Either create a new course or access an existing one
4. Navigate to the exercise overview
5. Create a quiz
6. Add Drag-And-Drop Question exercise with a background and a picture drag item to the quiz
7. Save the quiz
8. Navigate to the exercise overview
9. Click on import quiz
10. Import the created quiz as a new quiz
11. Click on Preview Quiz
12. Check, whether the imported exercise displays the background image and the picture drag item correctly

#### Exam Mode Testing



Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to the exam overview
3. Create a new exam
4. Navigate to the exercise group overview
5. Create a new exercise group
6. Import the previously created quiz
7. Navigate to the exam overview
8. Import the created exam as a new exam
9. Click on Preview Quiz
10. Check, if the exercise in the exported exam correctly displays the background image
11. Navigate to the exercise group overview
12. Import the exercise group from the firstly created exam.
13. Click on Preview Quiz
14. Check, if the exercise in the newly created exam correctly displays the background image

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage
